crates/sgx_validation/src/config.rs

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize};
+use sgx::parameters::Features;
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    /// Values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrsigner` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary which are disallowed.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_signer_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash: Option<Vec<Vec<u8>>>,
+
+    /// Prohibited values for `mrenclave` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_hex")]
+    pub enarx_hash_blacklist: Option<Vec<Vec<u8>>>,
+
+    /// Values for `features`.
+    #[serde(default)]
+    #[serde(deserialize_with = "from_features")]
+    pub features: Option<u64>,
```

Comment:
It seems that removing the `Option` means it's required to be in the config file, is that not the case?

Reply:
No, because you're using `serde(default)`, it would be simply `0` when not set, since `u64` implements `Default` https://doc.rust-lang.org/std/primitive.u64.html#impl-Default-for-u64

Reply:
Fixed